File: src/fizzbuzz.go
// FizzBuzz in Go
package main

import "fmt"

func main() {
	for i := 1; i <= 100; i++ {
		switch {
		case i%15 == 0:
			fmt.Println("fizzbuzz")
		case i%3 == 0:
			fmt.Println("fizz")
		case i%5 == 0:
			fmt.Println("buzz")
		default:
			fmt.Println(i)
		}
	}
}

File: src/fizzbuzz.hs
-- FizzBuzz in Haskell
module Main where
 
main :: IO ()
main = printAll $ map fizzBuzz [1..100]
       where
       printAll [] = return ()
       printAll (x:xs) = putStrLn x >> printAll xs
 
fizzBuzz :: Integer -> String
fizzBuzz n | n `mod` 15 == 0 = "fizzbuzz"
           | n `mod` 5  == 0 = "buzz"
           | n `mod` 3  == 0 = "fizz"
           | otherwise       = show n
